src/client/wire_protocol/operations.rs

```diff
@@ -662,23 +528,15 @@ impl Message {
     ///
     /// Returns the reply message on success, or an error string on
     /// failure.
-    pub fn read<T>(buffer: &mut T) -> Result<Message, String> where T: Read + Write {
-        let header = match Header::read(buffer) {
-            Ok(h) => h,
-            Err(s) => {
-                let str = format!("Unable to read reply header: {}", s);
-
-                return Err(str)
-            }
-        };
-
+    pub fn read<T>(buffer: &mut T) -> Result<Message> where T: Read + Write {
+        let header = try!(Header::read(buffer));
         match header.op_code {
             OpCode::Reply => {
                 Message::read_reply(buffer, header)
             },
             opcode => {
-                Err(format!("Expected to read response but instead found: {}",
-                            opcode.to_string()))
+                Err(ResponseError(format!("Expected to read response but instead found opcode {}",
```

Comment:
If you're going to include information about the specific opcode, you might as well also include `OpCode::Reply` in the error message.